content/tutorials/python-whirlwind-tour.md

```diff
@@ -1,720 +1,826 @@
 
-This concept is called 'scope'. We say that variables which are changed inside a function are in a different scope from those outside.
 
-You can have functions within functions, and this can actually be quite useful. In this situation, each nested function will also have its own scope.
 
-## Exercises: variables and mathematics
-### Average calculator
+* If Statements
+    * [Python Docs (if)](https://docs.python.org/3/tutorial/controlflow.html#if-statements)
+    * [Programiz (if - elif - else)](https://www.programiz.com/python-programming/if-elif-else)
+    * [W3Schools (if - elif - else)](https://www.w3schools.com/python/python_conditions.asp)
+    * [Tutorialspoint (if - elif - else)](https://www.tutorialspoint.com/python/python_if_else.htm)
+* While Loops
+    * [Tutorialspoint](https://www.tutorialspoint.com/python/python_while_loop.htm)
+    * [Realpython](https://realpython.com/python-while-loop/)
+    * [Programiz](https://www.programiz.com/python-programming/while-loop) 
+* For Loops
+    * [Python Docs](https://docs.python.org/3/tutorial/controlflow.html#for-statements)
+    * [Realpython](https://realpython.com/python-for-loop/)
+    * [Programiz](https://www.programiz.com/python-programming/for-loop)
+    * [W3Schools](https://www.w3schools.com/python/python_for_loops.asp)
 
-The first two lines of this program put two numbers entered by the user into variables `a` and `b`. (The `input` function is like `input`, but returns a number (e.g. `42`) when you enter one, rather than a string (like `"42"`).) Replace the comment with code that averages the numbers and puts them in a variable called `average`.
 
-```python
-a = input("Enter first number: ")
-b = input("Enter second number: ")
+## Doing Maths in Python
 
-# Store the average of a and b in the variable `average`
 
-print("The average of", a, "and", b, "is", average)
-```
+### Basic Operators
 
-Run your code and check that it works.
+Maths in Python is very similar to the way we do maths in a calculator. The order of operations follows _BIDMAS_ (_Brackets_, _Indices_, _Division_, _Multiplication_, _Addition_ and _Subtraction_). We would recommend using a liberal amount of brackets to ensure that the correct order of operations is carried out.
 
-### Distance calculator
+You can also use variables when doing mathematical operations, however they need to be floats or integers. Mathematical operations won’t always work on strings or booleans.
 
-Write a program which uses `input` to take an X and a Y coordinate, and calculate the distance from (0, 0) to (X, Y) using Pythagoras' Theorem. Put the code into an interpreter and run it. Does it do what you expected?
+Below is a list of mathematical operators that you can use:
 
-{{% notice tip %}}
-You can find the square root of a number by raising it to the power of 0.5, for example, `my_number ** 0.5`.
-{{% /notice %}}
 
-**Extension:** can you adapt the program to calculate the distance between any two points?
+<table>
```

Comment:
Hugo uses markdown tables instead of HTML tables:
https://www.markdownguide.org/extended-syntax/#tables

Reply:
Oof, i'll have to change that then

Reply:
#163